tuna.go

```diff
@@ -298,7 +298,10 @@ func (c *Common) CreateServerConn(force bool) error {
 			if subscribersCount == 0 {
 				return errors.New("there is no service providers for " + c.Service.Name)
 			}
-			offset := rand.Intn(int(subscribersCount))
+			offset := rand.Intn(subscribersCount)
+			if filter != nil && !filter.Empty() {
+				offset = 0
```

Comment:
This is incorrect: you will always get the first subscriber if filter is not empty.

Reply:
As I commented before, we should no longer getting one subscriber at a time because it's very inefficient. So L294-L308 should be changed:
1. Define a global constant, e.g. `getSubscribersBatchSize` to some value like 32
2. `offset := rand.Intn(int(subscribersCount/getSubscribersBatchSize) + 1)`
3. `subscribers, err := c.Wallet.GetSubscribers(topic, offset, getSubscribersBatchSize, true, false)`